src/usecase/feature_engineering.py

```diff
@@ -0,0 +1,294 @@
+    parameters
+    ----------
+    dataframe: pd.DataFrame
+        contextuals features are computed from the original dataset
+
+    returns
+    -------
+    dataframe: pd.DataFrame
+        dataframe with contextual features
+    """
+    
+    dataframe.timestamp = dataframe.timestamp.apply(convert_timestamp)
+    dataframe = dataframe.sort_values(by = 'timestamp').reset_index(drop=True)
+
+    dataframe['mean_diff'] = dataframe['mean_hr'].diff().shift(-1)
+    dataframe['mean_nni_diff'] = dataframe['mean_nni'].diff().shift(-1)
+    dataframe['max_hr_diff'] = dataframe['max_hr'].diff().shift(-1)
+
+    # time features
+    dataframe['month'] = dataframe.timestamp.dt.month
+    dataframe['dayOfWeek'] = dataframe.timestamp.dt.dayofweek
+    dataframe['hour'] = dataframe.timestamp.dt.hour
+    # dataframe['minute'] = dataframe.timestamp.dt.minute
+    # dataframe['second'] = dataframe.timestamp.dt.second
+
+    for period in [30, 60, 120]:
+
+        # Moving averages on different periods
+        dataframe['mean_nni_'+str(period)] = dataframe['mean_nni'].rolling(window=period, min_periods=10).mean()
+        dataframe["mean_hr_%s"%(period)] = dataframe['mean_hr'].rolling(window=period, min_periods=10).mean()
+        dataframe['max_hr_%s'%(period)] = dataframe['max_hr'].rolling(window=period, min_periods=10).max()
+        dataframe['min_hr_%s'%(period)] = dataframe['min_hr'].rolling(window=period, min_periods=10).min()
+
+        # Std on differents periods
+        dataframe['sdnn_%s'%(period)] = dataframe['sdnn'].rolling(window=period, min_periods=10).std()
+        dataframe['sdsd_%s'%(period)] = dataframe['sdsd'].rolling(window=period, min_periods=10).std()
+        dataframe['std_hr_%s'%(period)] = dataframe['std_hr'].rolling(window=period, min_periods=10).std()
+        dataframe['rmssd_%s'%(period)] = dataframe['rmssd'].rolling(window=period, min_periods=10).std()
+
+        dataframe['lf_%s'%(period)] = dataframe['lf'].rolling(window=period, min_periods=10).mean()
+        dataframe['hf_%s'%(period)] = dataframe['hf'].rolling(window=period, min_periods=10).mean()
+
+        # Std on differents periods
+        dataframe['vlf_%s'%(period)] = dataframe['vlf'].rolling(window=period, min_periods=10).mean()
+        dataframe['lf_hf_ratio_%s'%(period)] = dataframe['lf_hf_ratio'].rolling(window=period, min_periods=10).mean()
+
+        #non linear
+        dataframe['csi_%s'%(period)] = dataframe['csi'].rolling(window=period, min_periods=10).mean()
+        dataframe['Modified_csi_%s'%(period)] = dataframe['Modified_csi'].rolling(window=period, min_periods=10).mean()
+        dataframe['cvi_%s'%(period)] = dataframe['cvi'].rolling(window=period, min_periods=10).mean()
+    #     dataframe['sd1_%s'%(period)] = dataframe['sd1'].rolling(window=period, min_periods=10).mean()
+        dataframe['sd2_%s'%(period)] = dataframe['sd2'].rolling(window=period, min_periods=10).mean()
```

Comment:
j'extrairais ça dans une fonction avec "period" comme input, et je mettrais l'utilisation de cette fonction dans une liste compréhension. T'en penses quoi ? 
A la place de la boucle for 
[fonction(period) for period in list_periods]